Sources_v3/StreamChat/Database/DTOs/UserDTO.swift

```diff
@@ -18,7 +18,9 @@ class UserDTO: NSManagedObject {
     @NSManaged var userUpdatedAt: Date
     
     @NSManaged var flaggedBy: CurrentUserDTO?
-    
+    @NSManaged var currentUser: CurrentUserDTO?
+    var isCurrentUser: Bool { currentUser != nil }
```

Comment:
I'm very skeptical of this. In places where we needed this, we always fetched the instance of `CurrentUserDTO` and used it. Doesn't that approach work now?

Reply:
In this case, we access `UserDTO.isCurrentUser` not from db-session method (where `self` is NSMangedObjectContext) but from `ChatMessage.init(dto: MessageDTO)`.

We can jump around optional `dto.managedObjectContext` or use the `isCurrentUser` helper, but I'm not sure it's better keeping in mind that from performance perspective there is not difference because core-data is lazy and `currentUser` is fetched only when we access it.

Reply:
@VojtaStavik WDYT on that?

Reply:
I second to @b-onc here though my main concern is architecture-related. I just don't think it's a good idea to have a back reference from `User` to `CurrentUser`.

Reply:
We're using `dto.managedObjectContext!` in places already:
```swift
latestReplies = MessageDTO
            .loadReplies(for: dto.id, limit: 25, context: dto.managedObjectContext!)
            .map(_ChatMessage.init)
```
and
```swift
        let context = dto.managedObjectContext!
        
        let latestMessages: [_ChatMessage<ExtraData>] = MessageDTO
            .load(for: dto.cid, limit: 25, context: context)
            .map { $0.asModel() 
```
I believe we should use the same approach here

Reply:
-> also please see my comments regarding using fetch requests, i don't think we need anything like this

Reply:
fixed in 8ac7bb42aac85e9e420e5cfa0450d6536bfbda46